### DISCUSSION TITLE
Connector behavior

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#4045] [Q&A]
Hi everyone,

I've a strange issue when working with Connector : 
Consider the following world : 

<details>
<summary>world file</summary>
<pre>
#VRML_SIM R2021b utf8
WorldInfo {
}
Viewpoint {
  orientation -0.17089216477475663 -0.4038455301279104 -0.8987239041075431 3.5417107657060267
  position 0.9746144340677434 1.635663640808034 1.6395858652826367
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  rotation 1 0 0 1.5708
}
Robot {
  translation 0 0 0.15
  children [
    Connector {
      isLocked TRUE
      distanceTolerance 2
      axisTolerance 3.14
      rotationTolerance 3.14
      numberOfRotations 0
      snap FALSE
    }
  ]
  boundingObject Box {
    size 0.3 0.3 0.05
  }
  physics Physics {
  }
}
Solid {
  translation 0 0 0.32
  children [
    Connector {
      isLocked TRUE
      distanceTolerance 2
      axisTolerance 3.14
      rotationTolerance 3.14
      numberOfRotations 0
      snap FALSE
    }
  ]
  name "sphere"
  boundingObject Sphere {
    radius 0.04
  }
  physics Physics {
  }
}
</pre>
</details>

(In view option activate "show bounding box" to see anything :) )

First notice the two connectors in both the robot and the sphere, initially on locked state. Note also that their parameters allows them to lock in this configuration. According to the documentation, they should be linked and locked since the beginning of the simulation.
Run the simulation, the box and sphere fall, but the sphere is not locked with respect to the box.
I should be missing something here... 
Any idea ?

**Comment by ad-daniel** [comment]
It appears to be due to the fact that the connector axis of the sphere is pointing in the wrong direction. If you change its rotation field (on the sphere side) to `0 1 0 3.1415` it behaves correctly. You can display the connector axis under `view > optional rendering > show connector axis`.


https://user-images.githubusercontent.com/44834743/146523736-6df104f5-de84-4c63-8096-f9a6a27860d3.mp4

**Reply by ShuffleWire** [reply]
Daam ! Ok ! But why is that a thing ? I'm using
```
axisTolerance 3.14
rotationTolerance 3.14
```
that should allows to lock anyway, isn't it ?

**Reply by ShuffleWire** [reply]
I might have an idea, I'm using 3.14 as tolerance, but 3.14<pi :/ I'm right ? I've try using 3.141592653589793, but no more luck
Additionally, is there a way to disable the tolerance check ?

**Reply by ShuffleWire** [reply]
Ok got it i think ! 
We can't set 3.141592653589793 from the GUI, the value is cut to 3.14159 (after opening the wrl file, we can see that, hence when reopening it obviously fail) 
So I use 3.141592653589793 directly in the wrl and it work :)
